### PR TITLE
feat(appengine): dynamic credentials support (kork-credentials)

### PR DESCRIPTION
**Pull request by ncknt**
This moves the appengine provider to use `kork-credentials` abstractions.

This allows users to reload accounts dynamically, define new source of account information, or customize any part of the appengine credentials.

- `GoogleCommonManagedAccount` now extends `CredentialsDefinition`
- use `AbstractAtomicOperationsCredentialsConverter<>` instead of `AbstractAtomicOperationsCredentialsSupport`
- use `CredentialsRepository` instead of the `AccountCredentialsProvider`
- extend `BaseProvider` like Kubernetes provider
- Simplified agent handling with a credentials lifecycle handler

### CODE REVIEW

**Comment by ncknt** [comment]
@ezimanyi Could you PTAL?

File: clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/converters/AppengineAtomicOperationConverterHelper.groovy [hunk 1]
```diff
@@ -40,7 +41,7 @@ class AppengineAtomicOperationConverterHelper {
       .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
       .convertValue(input, targetDescriptionType)
 
-    converted.credentials = credentials in AppengineNamedAccountCredentials ? credentials : null
```
Comment (danielpeach) [inline clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/converters/AppengineAtomicOperationConverterHelper.groovy:43]: What on earth did `in` mean in this context?

File: clouddriver-appengine/src/main/java/com/netflix/spinnaker/clouddriver/appengine/config/AppengineCredentialsConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                        .sshTrustUnknownHosts(a.isSshTrustUnknownHosts())
+                        .gcloudReleaseTrack(a.getGcloudReleaseTrack())
+                        .services(a.getServices())
+                        .versions(a.getVersions())
+                        .omitServices(a.getOmitServices())
+                        .omitVersions(a.getOmitVersions())
+                        .cachingIntervalSeconds(a.getCachingIntervalSeconds())
+                        .build();
+                  } catch (Exception e) {
+                    log.info("Could not load account $managedAccount.name for App Engine", e);
```
Comment (danielpeach) [inline clouddriver-appengine/src/main/java/com/netflix/spinnaker/clouddriver/appengine/config/AppengineCredentialsConfiguration.java:100]: `$managedAccount` looks like a groovy holdover

**Review by danielpeach — APPROVED** [review]

File: clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/converters/AppengineAtomicOperationConverterHelper.groovy (hunk 1 above)
Comment (ncknt) [reply clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/converters/AppengineAtomicOperationConverterHelper.groovy:43]: Great question. I am not a groovy specialist but `credentials in AppengineNamedAccountCredentials` mysteriously returns true.

**Pull request merged**